tools/ARIAtools/extractProduct.py

```diff
@@ -294,22 +300,33 @@ def merged_productbbox(product_dict, workdir='./', bbox_file=None, croptounion=F
         # Generate footprint for the common intersection of all products
         else:
             prods_bbox=prods_bbox.intersection(total_bbox)
-        # Check if there is any common overlap
-        if prods_bbox.bounds==():
-            raise Exception('No common overlap, footprint cannot be generated. Last scene checked: %s'%(scene['productBoundingBox'][0]))
-        save_shapefile(prods_TOTbbox, prods_bbox, 'GeoJSON')
+            # Estimate percentage of overlap with bbox
+            if prods_bbox.bounds==():
+                print('Rejected scene %s has no common overlap with bbox'%(scene['productBoundingBox'][0]))
+                rejected_scenes.append(product_dict.index(scene))
+                os.remove(scene['productBoundingBox'][0])
+            else:
+                per_overlap=(prods_bbox.area/total_bbox.area)*100
+                # Kick out scenes below specified overlap threshold
+                if per_overlap<percentOverlap:
+                    print("Rejected scene %s has only %.2f%% overlap with bbox"%(scene['productBoundingBox'][0],per_overlap))
+                    rejected_scenes.append(product_dict.index(scene))
+                    os.remove(scene['productBoundingBox'][0])
+                else:
+                    save_shapefile(prods_TOTbbox, prods_bbox, 'GeoJSON')
+
+    # Remove scenes with insufficient overlap w.r.t. bbox
+    metadata_dict = [i for j, i in enumerate(metadata_dict) if j not in rejected_scenes]
+    product_dict = [i for j, i in enumerate(product_dict) if j not in rejected_scenes]
+    if product_dict==[]:
+        raise Exception('No common track overlap, footprints cannot be generated.')
 
     # If bbox specified, intersect with common track intersection/union
     if bbox_file is not None:
         user_bbox=open_shapefile(bbox_file, 0, 0)
         total_bbox=open_shapefile(prods_TOTbbox, 0, 0)
         user_bbox=user_bbox.intersection(total_bbox)
         save_shapefile(prods_TOTbbox, user_bbox, 'GeoJSON')
-
-        # Estimate percentage of overlap with bbox
-        per_overlap=((user_bbox.area)/(open_shapefile(bbox_file, 0, 0).area))*100
-        if per_overlap<50.:
-            print("WARNING: Common track extent only has %d%% overlap with bbox"%per_overlap+'\n')
```

Comment:
At the product stage once could keep it to only products that cover the bbox.
At the merge stage one could compare the threshold of coverage.
The latter is now removed.

Reply:
Yeah, in this function we still compare the overlap of the merged product, not of each individual product. The logic here has been moved up to lines ~305-314

Reply:
ok thanks for confirming. The comparison on the merged box only is good. Note that that is different from the explanation given in the  PR description.